### PR TITLE
Update env choice param

### PR DESCRIPTION
**Pull request by ilya-varchenya**

### CODE REVIEW

**Comment by ilya-varchenya** [comment]
image

File: src/com/qaprosoft/jenkins/jobdsl/factory/pipeline/PipelineFactory.groovy [hunk 1]
```diff
@@ -54,15 +54,8 @@ public class PipelineFactory extends JobFactory {
...
-        def envList = getGenericSplit(currentSuite, "jenkinsEnvironments")
-
-        if (envList.isEmpty()) {
-            envList.add("DEMO")
-            envList.add("STAG")
-        }
-
-        return envList
+    protected String getEnvironments(currentSuite) {
+        return currentSuite.getParameter("jenkinsEnvironments") == null || currentSuite.getParameter("jenkinsEnvironments").isEmpty() ? "DEMO\nPROD" : currentSuite.getParameter("jenkinsEnvironments")
```
Comment (vdelendik) [inline src/com/qaprosoft/jenkins/jobdsl/factory/pipeline/PipelineFactory.groovy]: please test that comma separate jenkinsEnvironments are splitted correctly and choice has 2+ options to select.
I suspect you have single option like: "DEMO, PROD"

File: src/com/qaprosoft/jenkins/jobdsl/factory/pipeline/PipelineFactory.groovy [hunk 2]
```diff
@@ -54,15 +54,20 @@ public class PipelineFactory extends JobFactory {
...
-        def envList = getGenericSplit(currentSuite, "jenkinsEnvironments")
-
-        if (envList.isEmpty()) {
-            envList.add("DEMO")
-            envList.add("STAG")
+    protected String getEnvironments(currentSuite) {
+        def enviroments = currentSuite.getParameter("jenkinsEnvironments")
+        def parsedEnviroments
+        if (enviroments != null && !enviroments.isEmpty()) {
+            if (!enviroments.contains(", ")) {
```
Comment (vdelendik) [inline src/com/qaprosoft/jenkins/jobdsl/factory/pipeline/PipelineFactory.groovy]: 1. use split by comma ","
2. trim items before adding
-> that's allow to support one-two-three etc spaces including tabs between

**Comment by okamara** [comment]
@ilya-varchenya what is expected result for the following cases? 
1. Case:
In the case when we delete Demo and write param -> <parameter name="jenkinsPipelineEnvironments" value="PROD, DEV"/> 
Actual result: DEMO STAG PROD appears
2. Case:
In the case when we delete Demo and write param ->
<parameter name="jenkinsPipelineEnvironments" value="STAG"/>
<parameter name="jenkinsEnvironments" value="DEMO"/>
<parameter name="jenkinsPipelineEnvironments" value="PROD, DEV"/>
Actual result: DEMO

**Pull request merged**